pkg/sql/lease.go

```diff
@@ -573,20 +584,57 @@ func (t *tableState) acquire(
 ) (*tableVersionState, error) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
-	// Wait for any existing lease acquisition.
-	t.acquireWait()
 
-	// Acquire a lease if no lease exists or if the latest lease is
-	// about to expire.
+	if t.mu.acquiring != nil {
+		// Block and wait for the new lease to be acquired on another routine.
+		t.acquireWait()
+	}
```

Comment:
You do not need to make this change because t.acquireWait() already checks if t.mu.acquiring != nil

Reply:
👍